Docs: Update GitLab SSO docs (#27853)
gravitational/teleport: docs/pages/access-controls/sso/gitlab.mdx

h1: Teleport SSO Authentication with GitLab
---

This guide will cover how to configure [GitLab](https://www.gitlab.com/) to issue
credentials to specific groups of users. When used in combination with role
based access control (RBAC), it allows administrators to define policies
like:

> - Only members of "DBA" group can access PostgreSQL databases.

Reviewer: ```suggestion
- Only members of the "DBA" group can access PostgreSQL databases.
```

- Only members of "ProductionKubernetes" can access production Kubernetes clusters
- Developers must never SSH into production servers.

## Prerequisites

- At least two groups in GitLab with users assigned. In our examples below,
  we'll assume a group named `company` with two subgoups, `admin` and `dev`.
- Teleport role with access to maintaining `oidc` resources.  This is available
  in the default `editor` role.

(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

- (!docs/pages/includes/tctl.mdx!)

## Configure GitLab

You should have at least one group configured in GitLab to map to Teleport roles.
In this example we use the names `gitlab-dev` and `gitlab-admin`.  Assign users
to each of these groups.

> 1. Create a Application in one of your Groups that will allow using GitLab as a

Reviewer: ```suggestion
1. Create an Application in one of your Groups that will allow using GitLab as a
```

> 1. Create a Application in one of your Groups that will allow using GitLab as a

Reviewer: Should "Application" and "Groups" be proper nouns here?

Author: No, the capitalization relates to the way they are presented in the GitLab nav menu. We could emphasize this with bold, or a screenshot, or just remove the caps.

Reviewer: I would be more explicit that "a Application" and "Groups" refer to the GitLab nav menu, otherwise this just looks like ungrammatical English. I think bolding the text and describing the relevant parts of the UI would be good.

   OAuth provider to Teleport.

   Settings:

   - Redirect URL: `https://<proxy url>/v1/webapi/oidc/callback`.
   - Check `Confidential`, `openid`, `profile`, and `email`.

   ![Create App](../../../img/sso/gitlab/gitlab-oidc-0.png)

> 2. Collect the `Application ID` and `Secret` in the Application. These will be

Reviewer: Should "Application" be a proper noun here?

Author: Again, that's how they're presented in the UI

>    used in the Teleport OIDC Auth Connector:

Reviewer: ```suggestion
   used in the Teleport OIDC auth connector:
```

For consistency with the rest of the docs, since we tend not to capitalize "auth connector".

   ![Collection Information](../../../img/sso/gitlab/gitlab-oidc-1.png)

> 3. Confirm the GitLab Issuer Address.

Reviewer: Same re: capitalization of "Issuer Address"

Author: ```suggestion
3. Confirm the GitLab issuer address.
```

>    For GitLab cloud the issue address is `https://gitlab.com`. This allows

Reviewer: ```suggestion
   For GitLab.com, the issuer address is `https://gitlab.com`. This allows
```
https://about.gitlab.com/blog/2014/04/18/gitlab-cloud-becomes-gitlab-com/

   Teleport to access the Open-ID configuration at
   `https://gitlab.com/.well-known/openid-configuration`. If you are self
   hosting the issuer address is the path to your GitLab instance.

## Configure Teleport

### Create a OIDC Connector

> Create a OIDC connector resource using `tctl`.

Reviewer: ```suggestion
Create an OIDC connector resource using `tctl`.
```

<Tabs>
<TabItem label="GitLab.com">
Replace the application ID and secret with the values from GitLab:

```code

> $ tctl sso configure oidc --preset gitlab \

Reviewer: Should we use `Var`s here?

--id <APPLICATION-ID> \
--secret <APPLICATION-SECRET> \
--claims-to-roles groups,company/admin,admin \
--claims-to-roles groups,company/dev,dev > oidc.yaml
```

</TabItem>
<TabItem label="Self-hosted">
Replace the application ID and secret with the values from GitLab, and replace
`https://gitlab.company.com` with the path to your self-hosted GitLab instance:

```code

> $ tctl sso configure oidc --preset gitlab \

Reviewer: Should we use `Var`s here?
